src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -307,34 +307,35 @@ public override object Clone()
                 this.logger.LogTrace("(-)[IS_IBD]");
                 return;
             }
-            
+
             //uint32_t nFetchFlags = GetFetchFlags(pfrom, chainActive.Tip(), chainparams.GetConsensus());
 
-            var send = new GetDataPayload();
-            foreach (InventoryVector inv in invPayload.Inventory.Where(inv => inv.Type.HasFlag(InventoryType.MSG_TX)))
-            {
-                //inv.type |= nFetchFlags;
+            var inventoryTxs = invPayload.Inventory.Where(inv => inv.Type.HasFlag(InventoryType.MSG_TX));
 
-                // TODO: This is incorrect, in blocks only mode we should just add to known inventory but not relay
-                if (this.isBlocksOnlyMode)                 
-                    this.logger.LogInformation("Transaction ID '{0}' inventory sent in violation of protocol peer '{1}'.", inv.Hash, peer.RemoteSocketEndpoint);
+            lock (this.lockObject)
+            {
+                foreach (var inv in inventoryTxs)
+                {
+                    this.filterInventoryKnown.Add(inv.Hash);
```

Comment:
Is it intentional that potentially different items are now added to the filter than before?

Reply:
BTC core adds all items to the known inventory filter not just the ones that are sent like our previous logic did.

Reply:
I'm not familiar with that, so if that is the case then of course this is fine and intentional